spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/outbound/MongoDbOutboundGatewayTests.java

```diff
@@ -0,0 +1,305 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mongodb.outbound;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+
+import java.util.List;
+import java.util.function.Function;
+
+import org.bson.Document;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.convert.MongoConverter;
+import org.springframework.data.mongodb.core.query.BasicQuery;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.integration.expression.FunctionExpression;
+import org.springframework.integration.mongodb.rules.MongoDbAvailable;
+import org.springframework.integration.mongodb.rules.MongoDbAvailableTests;
+import org.springframework.integration.support.MessageBuilder;
+import org.springframework.integration.test.util.TestUtils;
+import org.springframework.messaging.Message;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * @author Xavier Padró
+ * @since 5.0
+ */
+@ContextConfiguration
+@RunWith(SpringJUnit4ClassRunner.class)
+public class MongoDbOutboundGatewayTests extends MongoDbAvailableTests {
+	private static final String COLLECTION_NAME = "data";
+
+	private static final SpelExpressionParser PARSER = new SpelExpressionParser();
+
+	@Autowired
+	private BeanFactory beanFactory;
+
+	@Autowired
+	private MongoOperations mongoTemplate;
+
+	@Autowired
+	private MongoConverter mongoConverter;
+
+	@Autowired
+	private MongoDbFactory mongoDbFactory;
+
+	@Before
+	public void setUp() {
+		this.mongoTemplate.save(this.createPerson("Artem"), COLLECTION_NAME);
+		this.mongoTemplate.save(this.createPerson("Gary"), COLLECTION_NAME);
+		this.mongoTemplate.save(this.createPerson("Oleg"), COLLECTION_NAME);
+		this.mongoTemplate.save(this.createPerson("Xavi"), COLLECTION_NAME);
```

Comment:
`BulkOperations.insert(List<? extends Object> documents)` ?
Just for the knowledges extension.